sgkit/distance/metrics.py

```diff
@@ -0,0 +1,77 @@
+"""This module implements various distance metrics."""
+
+import numpy as np
+from numba import guvectorize
+
+from sgkit.typing import ArrayLike
+
+
+@guvectorize(["void(float64[:], float64[:], float64[:])"], "(n),(n)->()", fastmath=True)  # type: ignore
+def correlation(x: ArrayLike, y: ArrayLike, out: ArrayLike) -> None:
```

Comment:
These should have some more overloads, or at least `int8` since that would be the common input to them.